Move edxapp vars to app

jmaupetit opened the pull request:
## Purpose

This is the second step of our application driven migration, now targeting the `edxapp` application

## Proposal

- [x] fix app/services labels in templates
- [x] define an host+port for the patient0/development project
- [x] move `edxapp` variables from `group_vars/all/main.yml` to `apps/edxapp/vars/all/main.yml` and use them in templates
- [x] all synchronous sequential job execution :tada: 
- [x] test `edxapp`'s deployment in the CI
- [x] wait for all services to be up before running jobs
- [x] ~add support for multiple hosts per-service~ **edit**: postponed to another PR (see #90)
- [x] ~add lms host definition and test it~ **edit**: postponed to another PR (see #90)

jmaupetit committed:
✨(vars) use edxapp vars in edxapp templates

Move edxapp specific variables to the edxapp defaults and use them in
edxapp templates. edxapp templates have been cleanup for cross
applications consistency (labels, etc.).

jmaupetit committed:
✨(jobs) allow ordered sequential jobs execution

We are now able to run jobs in a particular order depending on their
template file name: jobs are sorted by their filename for an application
(all services), e.g.:

1. apps/edxapp/templates/cms/job_01_collectstatic.yml.j2
2. apps/edxapp/templates/lms/job_02_collectstatic.yml.j2
3. ...

And jobs are now executed sequentially, e.g. a new job cannot be
submitted if the previous one in the list did not complete.

jmaupetit committed:
🐛(circle) fix host vm/max_map_count to run ES

The CI VM needs to be able to run a dockerized ES instance so that
Richie's "regenerate index" job can run properly.

jmaupetit committed:
👷(circle) test edxapp(cms) bootstrapping

edxapp(cms) deployment is now also tested in the CI \o/

jmaupetit committed:
✨(jobs) wait for services before running jobs

To prevent jobs failure due to service unavailability, we've added a
wait loop that ensures that all services have been created first.

jmaupetit committed:
✨(scripts) update dev script to oc cluster

The bin/dev script now uses the "oc cluster" command to start a local
openshift cluster instead of minishift. This lowers down the
requirements and configuration time to start working with Arnold (and it
drastically improves the cluster performances).

The minishift start script has been kept and moved to bin/minishift for
those who would still prefer this legacy solution.

Fix #81

sampaccoud commented on apps/edxapp/templates/lms/svc.yml.j2:
```diff
@@ -2,17 +2,19 @@ apiVersion: v1
   labels:
-    app: edxapp-lms
-    version: "{{ edxapp_tag }}"
+    app: lms
```
should be edxapp?

jmaupetit replied:
Wow, well spotted!

sampaccoud commented on bin/dev:
```diff
@@ -2,88 +2,42 @@
 #
-# We use KVM as the default hypervisor, if you want to use VirtualBox instead,
-# please set the USE_KVM environment variable to 0, e.g.:
+# Berore running this script, please make sure that your system met the
```
s/Berore/Before

sampaccoud commented on bin/dev:
```diff
@@ -2,88 +2,42 @@
+#              "insecure-registries" : ["172.30.0.0/16"]
+#          }
+#
+#    2. To run ES, you will to ensure that your kernel's vm.max_map_count
```
you will NEED to ensure

sampaccoud commented on bin/dev:
```diff
@@ -2,88 +2,42 @@
-function _login() {
-    _set_minishift_path
-    oc login "https://$(minishift ip):8443" --username=developer --password=developer
+    oc cluster up --server-loglevel=5 --public-hostname="${OPENSHIFT_DOMAIN}"
```
that's a bit more than login!?

jmaupetit replied:
This function is no longer called login. Forget about this crappy diff.

sampaccoud commented on group_vars/all/main.yml:
```diff
@@ -32,8 +32,8 @@ trashable_env_types:
-cms_port:       8081
-lms_port:       8071
+# cms_port:       8081
+# lms_port:       8071
```
dead code

sampaccoud commented on group_vars/all/main.yml:
```diff
@@ -43,20 +43,20 @@ nginx_ports:
+# mongodb_image:       "centos/mongodb-32-centos7"
+# mongodb_tag:         "3.2"
+# mysql_image:         "centos/mysql-57-centos7"
+# mysql_tag:           "5.7"
```
dead code

sampaccoud commented on group_vars/all/main.yml:
```diff
@@ -85,10 +85,10 @@ lms_host: "{{ project_name }}-lms-{{ deployment_stamp }}.{{ domain_name }}"
+#   port: "27017"
+# mysql:
+#   host: "mysql"
+#   port: "3306"
```
dead code

jmaupetit committed:
⚡️(edxapp) add load mysql dump job

Running edxapp migrations is way too long (more than 20 minutes). Hence
we decided to speed up edxapp deployment by adding a new job that loads
a sql dump of the database schema before running migrations.

jmaupetit commented:
We still have a timeout from CircleCI's test-bootstrap job. We decided to merge this as-is and address this timeout issue in a new PR: the idea is to split the test-bootstrap job in a per-app job, to hopefully prevent job failure due to a timeout issue.
